### PR TITLE
70 even handler to search button

### PR DESCRIPTION
**Pull request by hanran-y**
## Details 
Linked the search button and input box with event handler in main. Fetching from API with query and save to local storage. 
Not implemented:
populate search result cards (not sure if Misha has already done it? Will check his pr)
navigate to search results page (router?)

## New Changes
- add event handler to search button to fetch recipes for query 
- add fetched recipes to local Storage (store Recipe to local with query as category name)
- add name "Search recipes input" to landing.html for getting input

## Linked Issue
Include issue number here
#70

## Checklist
- [x] I have requested 2+ code reviewers
- [ ] I have reviewed and commented my own code.
- [ ] My code follows the style guidelines of this project.
- [ ] I have added new tests as necessary.
- [ ] All new and existing tests pass successfully.

### CODE REVIEW

**Comment by hanran-y** [comment]
For API helpers: Only add keyword export, no changes. 
Will use New version API helpers once treeebooor finishes.

File: source/js/apiHelpers.js [hunk 1]
```diff
@@ -135,31 +135,31 @@ async function getRecipesByType(type, num) {
 // not sure how to do this
 
 
-/**
```
Comment (RainbowPangolin) [inline source/js/apiHelpers.js:138]: We need Robert to look at this change, as I have no idea how two versions of the same function snuck its way into the main branch and didn't give him any errors on his machine.

File: source/js/main.js [hunk 2]
```diff
@@ -96,3 +96,30 @@ function createExploreRecipeCards(storageCategoryData) {
...
 }
+
+
+/**
+ * active when search button click
+ * store Recipe to local with query as category name
+ * Does NOT apply any filter & sort, only search by name
+ * 1.add even handler to search button to fetch recipes for query 
+ * 2.add fetched recipes to local Storage
+ * 3.
```
Comment (RainbowPangolin) [inline source/js/main.js:107]: Please update the comments to be complete. The trailing '3.' concerns me, as that signals to me that you've missed something. 

Further, put a TODO for anything that needs to be done in the future.

File: source/js/main.js [hunk 3]
```diff
@@ -96,3 +96,30 @@ function createExploreRecipeCards(storageCategoryData) {
...
+ */
+async function searchButtonHandler() {
+  let query = document.querySelector('input[name="Search recipes input"]').value;
+  let num = 2;    //hard coded, set small num for testing
+
+  let searchResult = await apiFuncs.getRecipesByName(query, num);
+  // console.log("data from API");
+  // console.log(searchResult);
```
Comment (RainbowPangolin) [inline source/js/main.js:115]: Remove commented out console logs.

**Review by RainbowPangolin — CHANGES_REQUESTED** [review]
Code needs to be better commented first, and I will add Robert to this so he can see that apiHelpers.js is broken in the main branch. Other than that, good start

File: source/js/apiHelpers.js (hunk 1 above)
Comment (RainbowPangolin) [reply source/js/apiHelpers.js:138]: Resolved in #102

**Comment by RainbowPangolin** [comment]
Summary of changes requested:

1. Clean up comments
2. Use the new version of apiHelpers.js (We can just use the manual merge-conflict tool on GitHub)

**Comment by hanran-y** [comment]
Going update(merge) main to my branch, I will submit a new pr.

**Pull request closed without merging**